ray-curator/ray_curator/stages/audio/inference/asr_nemo.py

```diff
@@ -0,0 +1,145 @@
+import time
+from dataclasses import dataclass, field
+from typing import ClassVar
+
+import nemo.collections.asr as nemo_asr
+import torch
+from loguru import logger
+
+from ray_curator.backends.base import NodeInfo, WorkerMetadata
+from ray_curator.stages.base import ProcessingStage
+from ray_curator.stages.resources import Resources
+from ray_curator.tasks import DocumentBatch, FileGroupTask, SpeechObject
+
+
+@dataclass
+class InferenceAsrNemoStage(ProcessingStage[FileGroupTask | DocumentBatch | SpeechObject, SpeechObject]):
+    """Stage that do speech recognition inference using NeMo model.
+
+    Args:
+        model_name (str): name of the speech recognition NeMo model. See full list at https://docs.nvidia.com/nemo-framework/user-guide/latest/nemotoolkit/asr/all_chkpt.html
+        filepath_key (str): which key of the data object should be used to find the path to audiofile. Defaults to “audio_filepath”
+        pred_text_key (str): key is used to identify the field containing the predicted transcription associated with a particular audio sample. Defaults to “pred_text”
+        cuda (str): device to run inference on it. Could be cpu, cuda, ipu, xpu, mkldnn, opengl, opencl, ideep, hip, ve, fpga, maia, xla, lazy, vulkan, mps, meta, hpu, mtia or cuda number (digit). Defaults to “” (empty string)
+        name (str): Stage name. Defaults to "ASR_inference"
+    """
+
+    model_name: str
+    filepath_key: str = "audio_filepath"
+    pred_text_key: str = "pred_text"
+    cuda: str = ""
+    name: str = "ASR_inference"
+    _start_time = time.time()
+    _metrics: ClassVar[dict] = {}
```

Comment:
These can be removed in favor of storing it in `_stage_perf`, right?

Reply:
removed